pkg/cluster/storageclass.go

```diff
@@ -12,11 +12,17 @@ import (
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
 const (
 	defaultStorageClassName          = "managed-premium"
 	defaultEncryptedStorageClassName = "managed-premium-encrypted-cmk"
+	defaultProvisioner               = "kubernetes.io/azure-disk"
+	ocp411StorageClassName           = "managed-csi"
+	ocp411EncryptedStorageClassName  = "managed-csi-encrypted-cmk"
+	ocp411Provisioner                = "disk.csi.azure.com"
```

Comment:
Can we make this more generic? Will we have to add more constants for 4.12 support? 4.13 support?

One of the patterns we used in Assisted Installer is to keep JSON files for the different versions to handle defaults see: https://github.com/openshift/assisted-service/blob/master/data/default_os_images.json

In Assisted Installer these are loaded at runtime instead of compile time. We could use this pattern and modify it for compile time if ARO needs it. Just thinking about maintainability of multi-version in the future. WDYT?

Reply:
I like the changes to make it more generic. :)

> They may change, but we are going to have to adapt the RP to handle any changes as we GA new versions, which means making version specific tweaks to our code in order to maintain multi-version compatibility. If 4.12 keeps the same storage class as 4.11, then we could remove the special handling when we have 4.11 and 4.12 as install options.

That's the pattern I was seeing if we have multiple versions of OCP in ARO for install we may end up with very complicated logic to deal with defaults that could be hard to back out. Different versions of OCP will tend to have different installation defaults. For now I think this change is good! 

It has got me thinking about a more proactive approach (external definitions or not) rather than reactive, to try to control the different settings for versions of OCP in ARO.